extensions/support/jackson-dataformat-xml/deployment/src/main/java/org/apache/camel/quarkus/support/jackson/datafromat/xml/deployment/JacksonDataformatXmlSupportProcessor.java

```diff
@@ -0,0 +1,80 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.support.jackson.datafromat.xml.deployment;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import javax.xml.stream.XMLEventFactory;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLOutputFactory;
+
+import com.ctc.wstx.shaded.msv.org_isorelax.verifier.VerifierFactoryLoader;
+import com.ctc.wstx.shaded.msv.relaxng_datatype.DatatypeLibraryFactory;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.ObjectCodec;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.DeploymentClassLoaderBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+import io.quarkus.deployment.util.ServiceUtil;
+import org.codehaus.stax2.validation.XMLValidationSchemaFactory;
+
+public class JacksonDataformatXmlSupportProcessor {
+
+    private static final String FEATURE = "camel-support-jackson-dataformat-xml";
+    static final String SERVICES_PREFIX = "META-INF/services/";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    void serviceProviders(BuildProducer<ServiceProviderBuildItem> serviceProviders,
+            final DeploymentClassLoaderBuildItem classLoader) {
+        Stream.concat(
+                Stream.of(
+                        JsonFactory.class,
+                        ObjectCodec.class,
+                        VerifierFactoryLoader.class,
+                        DatatypeLibraryFactory.class,
+                        XMLEventFactory.class,
+                        XMLInputFactory.class,
+                        XMLOutputFactory.class)
+                        .map(Class::getName),
+                Stream.of(
+                        XMLValidationSchemaFactory.INTERNAL_ID_SCHEMA_DTD,
+                        XMLValidationSchemaFactory.INTERNAL_ID_SCHEMA_RELAXNG,
+                        XMLValidationSchemaFactory.INTERNAL_ID_SCHEMA_W3C,
+                        XMLValidationSchemaFactory.INTERNAL_ID_SCHEMA_TREX)
+                        .map(schemaId -> XMLValidationSchemaFactory.class.getName() + "." + schemaId))
```

Comment:
As there's no discovery but we deal with a fixed set of ites, maybe we can use strings so we can avoid the the two `Stream.of` and the `Stream.concat`

Reply:
Yes, your proposal is valid and I can see the advantages but classes are easier to type in an IDE and any typos will cause a compilation error so it feels safer. I am open to discuss this.

Reply:
Yes it is not really related to perf of loading classes but more about about the `complexity` of two nested streams with mapping like looking at the code is seems much more complex that what it actually does but there's no problem to keep it as it is

Reply:
I am at least hanging this quote from Plato's Timaeus above my table to remind me not to sin against the beauty in the future ;)

> The good, of course, is always beautiful, and the beautiful never lacks proportion.